enterprise/internal/codeintel/autoindexing/internal/inference/lua/ruby.lua

```diff
@@ -0,0 +1,39 @@
+local path = require("path")
+local recognizer = require("sg.autoindex.recognizer")
+local pattern = require("sg.autoindex.patterns")
+
+local indexer = require("sg.indexes").get("ruby")
+local outfile = "index.scip"
+
+local pattern_list = {
+  pattern.new_path_extension("gemspec"),
+  pattern.new_path_basename("rubygems-metadata.yml"),
+  pattern.new_path_basename("Gemfile.lock"),
+  pattern.new_path_basename("Gemfile"),
+}
+
+return recognizer.new_path_recognizer {
+  patterns = pattern_list,
+
+  generate = function(api, paths)
+    local roots = {}
+    for i = 1, #paths do
+      roots[path.dirname(paths[i])] = true
+    end
+    for root in pairs(roots) do
+      api:register(recognizer.new_path_recognizer({
+        patterns = pattern_list, -- unused; only present to trigger generate
+        generate = function(api, _paths)
+          return {
+            steps = {},
+            root = root,
+            indexer = indexer,
+            indexer_args = { "scip-ruby-autoindex" },
+            outfile = outfile,
+          }
+        end,
+      }))
+    end
+    return {}
```

Comment:
```suggestion
    local jobs = {}
    for root in pairs(roots) do
      table.insert(jobs, {
        steps = {},
        root = root,
        indexer = indexer,
        indexer_args = { "scip-ruby-autoindex" },
        outfile = outfile,
      })
    end
    return jobs
```

Is there a reason you're registering a new callback here that isn't conditional? You can just return a list of jobs.

Reply:
It wasn't obvious what the return value was meant to be -- the other files seem to be returning empty lists in a bunch of places.